server/server.go

```diff
@@ -46,7 +47,13 @@ func Run(ctx context.Context, listenAddr string, logger *zap.Logger) {
 
 func setupRouter(logger *zap.Logger) *gin.Engine {
 	router := gin.New()
-	router.Use(ginzap.Ginzap(logger, time.RFC3339, true))
+
+	router.Use(middleware.Logging(logger))
+	router.Use(middleware.Recovery)
+
 	router.GET("/", hello)
+	router.GET("/error", err)
```

Comment:
I'd recommend making a group `example` and move all this handler to this one group